### ISSUE TITLE
In-App Message: JSON Feature Flags

### ISSUE THREAD

**Issue by joethreepwood** [PostHog/meta#100]
## TL;DR
I want to trial a targeted in-app prompt to test how effective they are at driving users to adopt new features. From discussion with @lharries and @annikaschmid JSON feature flags themselves are a good candidate for a feature we can drive users towards, because they're powerful and unique but [have had low adoption so far.](https://app.posthog.com/insights/Vv7HjlFf) 

## Targeting
I'm proposing we target all users who have created or updated a FF in the last 30 days (there's [900ish of them](https://app.posthog.com/cohorts/28306)) with a simple message, triggered through the in-app prompts. It'd be a one-time thing to draw attention to JSON FFs. 

## Message
A short pop-up prompt in the bottom right corner of the screen, with the Hog image below.

![posthog-array-blog](https://user-images.githubusercontent.com/84011561/223414954-48e9603c-d6d4-4ae3-9497-92652c7e7fc2.jpg)
**Now available: JSON feature flags**
You can now send arbitrary data as a JSON payload on flags, enabling you to make real-time product changes without redeploying!
CTA: Try it now

We'd send users to https://app.posthog.com/feature_flags/new. An alternate CTA would be to send users to the docs. Open to feedback on this.

**Comment by lharries** [comment]
Agreed of this being a good idea for an experiment

Heads up that the button link currently opens in a new tab, pretty easy to change this however if we want.

Most people won't know what JSON feature flags are or why they should use them. The modal has quite a bit of space so might want to add more context here? Or have the picture be a code block etc.

**Comment by lharries** [comment]
Btw we can also make this only show on the feature flags page, which would get a bit less reach but be more timely?

**Comment by neilkakkar** [comment]
Love the idea!

I'd like it if this happens towards the end of this sprint vs the beginning though, payloads have been in beta, and there's some confusion around how to use them in code we're focusing on solving this sprint (@liyiy is working on this), and would be nice to get it in before this goes out.

**Comment by joethreepwood** [comment]
<img width="382" alt="Screenshot 2023-03-10 at 12 26 03" src="https://user-images.githubusercontent.com/84011561/224315653-7c003377-874e-44a6-88a1-3d72abf8edc9.png">

Some changes, but this is what I'm planning to roll out later today.

**Comment by jamesefhawkins** [comment]
The way we upsell/crossell will get increasingly important. I think we'll often face a painful tradeoff between brand damage and long run slower growth (fewer recommendations) versus better commercial performance.

I am genuinely curious why the existing notification system is not good enough?
<img width="54" alt="Screenshot 2023-03-10 at 13 56 52" src="https://user-images.githubusercontent.com/47497682/224334369-e1cbaaa2-a8b7-4e96-977d-ac819969f600.png">

Whilst I think this has a good chance of hitting your goal, do we want a product that does this for everything that might be relevant to our users in multiple places?

Should our changes to the product always be the above instead / or is this too noisy? Why then would we do it for this one specific thing?

**Comment by raquelmsmith** [comment]
I somewhat agree with James about popups. Most of the time I find them annoying, mostly because they usually aren't relevant to what I'm trying to do _right now_. It would be interesting to do a multivariate test here - one with the popup, one with a notification, and one with a lil tooltip/tag on the payload field itself to call attention to it. See which ones gets people to use it more. Not sure if we have the volume to see results in a reasonable amount of time for that though...

**Comment by joethreepwood** [comment]
I'm actually interested in testing notifications as well, and have discussed using them for this to promote the changelog. My understanding though was that this would need [some code changes](https://github.com/PostHog/posthog/pull/14572) which @pauldambra was looking at.  

Very open to feedback on this, but my thinking for this split was that notifications were less intrusive and more missable, so should be used for content which is less targeted or more regular. In-app popups are more intrusive, so should be used in a more targeted way to give more actionable messages. 

Therefore, use notifications for the changelog every week; use in-app popups to drive adoption at a targeted group. 

> It would be interesting to do a multivariate test here - one with the popup, one with a notification, and one with a lil tooltip/tag on the payload field itself to call attention to it.

Yeah, definitely not enough volume here. Though, to your other point we could deliver this pop-up only on the Feature Flag page as Luke suggested -- that'd help solve the contextual issue you described. 

Either way, we're currently delaying for a week after discussion with Neil based on some upcoming FF changes.

**Comment by lharries** [comment]
Agreed that we don't want it to be spammy but we still want to drive feature adoption. I think this comes down to design and showing in the right context.

Here was a recent good example from Github and resulted in positive word of mouth internally from Michael. Related to targeting - if you have never used github comments there would be no need to show the tooltip as you wouldn't be aware that it was missing that functionality.

> Therefore, use notifications for the changelog every week; use in-app popups to drive adoption at a targeted group.

I'd therefore propose something like: "Use the **notifications section** for the changelog every week; use in-app **tooltips** to drive adoption at a targeted group for larger features.". For targeting we generally wouldn't want them for new users.

There's also the setting which is max 1 prompt per day. Which we could increase to 1 prompt per week etc.

<img width="757" alt="Screenshot dev - PostHog - 11 new items - Slack (Slack) 2023-03-14 at 12 00@2x" src="https://user-images.githubusercontent.com/22766134/224994684-4da339fd-c53a-404a-a90c-e8d4897901ea.png">

@raquelmsmith is this something you'll be wanting to iterate with on the growth side and would therefore work with Joe on? Otherwise I work on a better tooltip for people to use as default instead of the bottom-right popup

**Comment by raquelmsmith** [comment]
Oh yes, I love how github did theirs. I also saw it and thought "this is great!" but dismissed it before I remembered to take a screenshot. I think I wasn't super explicit about my dislike for popups - I don't like popups/modals that show up randomly to announce something unrelated to what I'm doing. I love contextual, well-timed popups that help me use the product _right now_. The github one is a really good example. 

> @raquelmsmith is this something you'll be wanting to iterate with on the growth side and would therefore work with Joe on? Otherwise I work on a better tooltip for people to use as default instead of the bottom-right popup

I would love to own this but my plate is currently full so I can't in good faith take this on right now 😅  But - if you can make it targeted to a certain element and do a tooltip style thing like the github example then that would be fantastic and good enough for Joe to move forward with.

**Comment by joethreepwood** [comment]
I think, between targeting to cohorts and other existing features, the only improvement we'd need is to make it show on specific pages. We can already change the position and target it to particular users, and what I'm hearing here is that this would fit a lot better if we limited it to appearing on the Feature Flags page.

**Comment by joethreepwood** [comment]
With help from @pauldambra, we'll now use in-app notifications in the notification bar for the changelog going forward. 

https://github.com/PostHog/posthog/pull/14572

**Comment by joethreepwood** [comment]
I've now updated the feature flag to only show on the feature flag page: https://app.posthog.com/feature_flags/8396

**Comment by joethreepwood** [comment]
Tracking the adoption of JSON vs regular FFs [here](https://app.posthog.com/insights/Vv7HjlFf) and, after turning the flag on earlier this week, it does look like we've had a small uptick in the amount of usage. And 0 complaints. 

It's a really tiny test admittedly, but an encouraging sign for doing a bigger test.
